src/lib/cbuff/cbuff_datatypes.h

```diff
@@ -0,0 +1,25 @@
+/**
+ * @file cbuff_datatypes.h
+ * @author Salvador Z
+ * @version 1.0
+ * @brief File for Circular/ring buffer datatypes definition
+ *
+ */
+
+#ifndef CBUFF_DATATYPES_H_
+#define CBUFF_DATATYPES_H_
+
+// Includes
+#include "utils_common.h"
+
+typedef struct cbuff_s {
+  void    *vBuff;    // Will hold the buffer ref
+  uint8_t  u8_eSize; // Element size
+  bool_t   b_isFull; // if true the buffer is full
```

Comment:
used to avoid wasting one slot on the cbuff

Reply:
Might cause a problem in multi-thread this flag would require a lock to avoid race conditions.